### PR TITLE
Create a reusable component in Upcoming Events section

### PR DESCRIPTION
**Pull request by shami-dev**
Issue: #35 #14 

## Summary
- Created a reusable EventCard component and used it inside EventsSection.
- Added imgUrl, imgAlt, and location (optional) to Prisma's Events model. 

## Details (reason and description of the changes)
EventsSection function can receive "events" and "groups" arrays and render cards using the EventCard component. imgUrl and imgAlt rows are needed since we are using them in UI. 

Important: Don't forget to re-run the Prisma generate command since the Prisma schema has changed. 

### Challenges (what problems did I face and how did I fix them?)
Generating the second card in the Upcoming Events section. 
Resolved with ChatGPT) It suggested using indexes to generate an image of the second card on the bottom.

## Screenshots
On the desktop. 
Screenshot 2023-08-10 at 11 54 31 AM
On the mobile.
Screenshot 2023-08-10 at 11 56 22 AM

### CODE REVIEW

**Review by tonydangblog — COMMENTED** [review]
I left some comments. Lmk if you think making `EventPreview` a single div is a good idea or if you would like to keep things as-is.

File: app/routes/_index.tsx
Comment (andrelandgraf) [inline app/routes/_index.tsx:22]: Curious, why did you decide to declare the component prop types here? I haven't seen this pattern before. I would just keep them in the same file as the component. This keeps the types nicely co-located with its usage.

File: app/routes/_index.tsx
Comment (andrelandgraf) [inline app/routes/_index.tsx:9]: In the last tutoring session, we defined the event schema in the `prisma.schema` file. Prisma generates TypeScript types from these schemas, which we can use in our code. Maybe you can use the auto-generated types here instead?

`import type { Event, Group, User } from '@prisma/client';`

File: app/routes/_index.tsx
Comment (andrelandgraf) [inline app/routes/_index.tsx:81]: Commented out code should probably be removed!

**Review by andrelandgraf — COMMENTED** [review]
Great work!

File: app/components/events-section.tsx
Comment (shami-dev) [reply app/components/events-section.tsx:1]: Oh, sure) Will fix it!

File: app/routes/_index.tsx
Comment (shami-dev) [reply app/routes/_index.tsx:22]: I thought we will use these types in other routes like search.tsx. That's why kept all types in one place. This pattern was more like an experiment and it worked. Moving types to the components sound reasonable. Thanks. I'll do it!

File: app/routes/_index.tsx
Comment (shami-dev) [reply app/routes/_index.tsx:81]: Oops, yes!

File: app/routes/_index.tsx
Comment (shami-dev) [reply app/routes/_index.tsx:9]: I think using Prisma types is a good idea. Could we include imgUrl and imgAlt in the Event model in schema.prisma? Also, should we add a 'location' column to the model if we plan on having events in different cities or countries?

File: app/components/events-section.tsx
Comment (shami-dev) [reply app/components/events-section.tsx:16]: I agree. This would make EventPreview more reusable. Thanks for the idea! Will do.

**Comment by shami-dev** [comment]
> I left some comments. Lmk if you think making `EventPreview` return a single div is a good idea or if you would like to keep things as-is.

It sounds great!

File: app/routes/_index.tsx
Comment (andrelandgraf) [reply app/routes/_index.tsx]: Oh yeah for sure! :) Feel free to add those fields!

**Review by tonydangblog — APPROVED** [review]
Approved and left some comments for future improvements!

File: app/components/events-section.tsx
Comment (willbeaumont) [reply app/components/events-section.tsx:28]: Future enhancement might be for the mobile view to not reorder the cards too. Having the format change while scrolling might look strange.

File: prisma/schema.prisma
Comment (willbeaumont) [reply prisma/schema.prisma:59]: This might be something good for a broader discussion. Are we going to require events to have images (if I understand this correctly)?

**Review by willbeaumont — APPROVED** [review]

**Review by willbeaumont — DISMISSED** [review]
Nice work!

File: prisma/schema.prisma
Comment (andrelandgraf) [reply prisma/schema.prisma:59]: Once we build the UI, we will have a clearer image (pun intended) of what fields we want to be optional and which ones are required. For now, I would recommend making all three new fields optional!

**Review by andrelandgraf — APPROVED** [review]
Tip: Once you have 2 approvals, try to merge asap to avoid further merge conflicts and re-review iterations. Focus on fixing the current conflicts by merging the latest `main` into your feature branch, and then we get this merged!

File: app/components/events-section.tsx
Comment (shami-dev) [reply app/components/events-section.tsx:3]: Yeah, we can leave it without exporting since use it only here.

File: prisma/schema.prisma
Comment (shami-dev) [reply prisma/schema.prisma:59]: Agree with @willbeaumont, it seems we can have a bunch of edge cases which is a great topic to discuss in a session. Didn't go so deep) Just thought about online events when made this field optional. However, would be great to have images for events to make our UI more appealing. At least we could use a group image if there are no event images. Ok @andrelandgraf. Will keep in mind that it should be optional fields!

File: app/components/events-section.tsx
Comment (shami-dev) [reply app/components/events-section.tsx:28]: @tonydangblog, that's a great idea. I'm going to use it!

Comment (shami-dev) [reply app/components/events-section.tsx:28]: @willbeaumont good catch! That is easy to implement 'sm:order-2'. Fix it!

**Comment by willbeaumont** [comment]
lgtm!

**Review by willbeaumont — APPROVED** [review]

**Review by andrelandgraf — APPROVED** [review]
Great work! LGTM!

**Pull request merged**
